src/main/java/com/aws/greengrass/deployment/KernelUpdateDeploymentTask.java

```diff
@@ -141,4 +161,9 @@ private DeploymentException getDeploymentStatusDetails() {
 
         return new DeploymentException(deployment.getStageDetails(), errorStack, errorTypes);
     }
+
+    @Override
+    public void cancel() {
+        deploymentResultCompletableFuture.cancel(false);
+    }
```

Comment:
We're already canceling the `deploymentResultCompletableFuture` in `DeploymentTaskMetada`. No need to have the `cancel` method here to do the same.

Reply:
The cancel function in DeploymentTaskMetadata calls this cancel function first to cancel the CompletableFuture for waitForServicesToStart and the Future that waits for KernelUpdateDeployment to return the deployment results.

It's two different futures, both needs to be cancelled for KernelUpdateDeployment